url: add value argument to has and delete methods (#47885)
nodejs/node: doc/api/url.md

Append a new name-value pair to the query string.

#### `urlSearchParams.delete(name, value)`

> 

Author: @anonrig @debadree25 
Made the doc changes.

Had couple of doubts.

i) how to show that `value` is optional argument. 
ii) Do I need to mention something like "added in version x.x.x". I saw something mentioned [here](https://github.com/nodejs/node/blob/main/doc/contributing/pull-requests.md#step-3-code)

<!-- YAML
added: REPLACEME
-->

Reviewer: i) to show wrap it in [ something like ```urlSearchParams.delete(name[, value])```
ii) Yes

Append a new name-value pair to the query string.

#### `urlSearchParams.delete(name[, value])`

<!-- YAML
changes:
  - version: REPLACEME
    pr-url: https://github.com/nodejs/node/pull/47885

>     description: added optional value argument.

Reviewer 2: ```suggestion
    description: Add support for optional `value` argument.
```

-->

* `name` {string}
* `value` {string}

> If `value` is provided, then delete all name-value pairs with

Reviewer 2: The phrasing should be consistent between the two paragraphs.
```suggestion
If `value` is provided, removes all name-value pairs with
```

Author: thanks

> whose `name` is name and `value` is value.

Reviewer 2: ```suggestion
whose name is `name` and value is `value`.
```

> If `value` not provided, remove all name-value pairs whose name is `name`.

Reviewer: ```suggestion
If `value` is not provided, remove all name-value pairs whose name is `name`.
```

> If `value` is not provided, remove all name-value pairs whose name is `name`.

Reviewer 2: ```suggestion
If `value` is not provided, removes all name-value pairs whose name is `name`.
```

Returns the values of all name-value pairs whose name is `name`. If there are
no such pairs, an empty array is returned.

#### `urlSearchParams.has(name[, value])`

<!-- YAML
changes:
  - version: REPLACEME
    pr-url: https://github.com/nodejs/node/pull/47885

>     description: added optional value argument.

Reviewer 2: ```suggestion
    description: Add support for optional `value` argument.
```

-->

* `name` {string}
* `value` {string}
* Returns: {boolean}

Checks if the `URLSearchParams` object contains key-value pair(s) based on
`name` and an optional `value` argument.

> If `value` is provided, returns true when name-value pair with

Reviewer 3: To be consistent with the next sentence:

```suggestion
If `value` is provided, returns `true` when name-value pair with
```
